### PR TITLE
Porting Validation for UI

### PR DESCRIPTION
**Pull request by ailZhou**
### Description

Grabbed validation code for reports from MCR. Made some code adjustments for certain report types so that it will use saved data.

DateField is fixed, was missing autosave.

@ntsummers1 has graciously agreed to take over this ticket while I'm out.

### Related ticket(s)

MDCT-2954

---
### How to test

1. Go around and just add/edit/delete

### Important updates

N/A

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by britt-mo — APPROVED** [review]
Worked as expected in the browser

File: services/app-api/forms/wp.json [hunk 1]
```diff
@@ -914,7 +913,7 @@
             "deleteModalConfirmButtonText": "Yes, delete objective",
             "deleteModalWarning": "You will lose all information entered for this objective. Objective will not be deleted from archived Work Plans and Semi-Annual Reports, but it will not be available in future Work Plans and Semi-Annual Reports.",
             "enterEntityDetailsButtonText": "Edit",
-            "dashboardTitle": "Objective total count: 2",
+            "dashboardTitle": "Objective total count",
```
Comment (karla-vm) [inline services/app-api/forms/wp.json:916]: Do we need a `countEntitiesInTitle: true` flag here?

**Review by karla-vm — DISMISSED** [review]

File: services/ui-src/src/components/modals/DeleteEntityModal.tsx [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
-    console.log("Entity Deleted", selectedEntity);
+    setDeleting(true);
+    const reportKeys = {
+      reportType: report?.reportType,
+      state: report?.state,
+      id: report?.id,
+    };
+    const currentEntities = report?.fieldData?.[entityType] || [];
+    const updatedEntities = currentEntities.filter(
+      (entity: EntityShape) => entity.id != selectedEntity?.id
```
Comment (britt-mo) [inline services/ui-src/src/components/modals/DeleteEntityModal.tsx:30]: is the single `=` intentional?

File: services/app-api/forms/wp.json (hunk 1 above)
Comment (ntsummers1) [reply services/app-api/forms/wp.json:916]: Good catch. Never knew about this field. Added!

File: services/ui-src/src/components/modals/DeleteEntityModal.tsx (hunk 2 above)
Comment (ntsummers1) [reply services/ui-src/src/components/modals/DeleteEntityModal.tsx:30]: Just a straight copy from MCR. I can make it double!

**Review by britt-mo — APPROVED** [review]

**Pull request merged**